packager/media/chunking/chunking_handler.h

```diff
@@ -72,7 +72,7 @@ class ChunkingHandler : public MediaHandler {
   // Segment and subsegment duration in stream's time scale.
   int64_t segment_duration_ = 0;
   int64_t subsegment_duration_ = 0;
-
+  int64_t set_segment_index_ = 0;
```

Comment:
There is already a variable for segment index: current_segment_index_. Why do we need another one?

Reply:
current_segment_index_ is used to compare with segment index calculated at line: https://github.com/google/shaka-packager/blob/8913dbda85967b26e4968495b1d0ed70fbffe1fd/packager/media/chunking/chunking_handler.cc#L99 and 
https://github.com/google/shaka-packager/blob/8913dbda85967b26e4968495b1d0ed70fbffe1fd/packager/media/chunking/chunking_handler.cc#L103
whereas
set_segment_index_ is used for keeping the segment count for file creation and needs to be incremented in onCueEvent and ChunkingHandler::OnFlushRequest (which also call EndSegmentIfStarted()). If the current_segment_index_ is used here instead of set_segment_index_, the result of comparison in https://github.com/google/shaka-packager/blob/8913dbda85967b26e4968495b1d0ed70fbffe1fd/packager/media/chunking/chunking_handler.cc#L103 will change.

Please let me know what you think.

Also I noticed that isGreaterSegmentIndex should use set_segment_index_ instead of current_segment_index_ (as set_segment_index_ should never be decremented). I will make this change.

Reply:
Good point! I am afraid that won't work though as the increment is undone by the code from line 109 to 113. 

Since the cue basically resets segment index back to 0 [1], I think we can keep a record of segment count: `num_segments_before_last_cue_`.

In `OnCueEvent` function:

```c++
Status ChunkingHandler::OnCueEvent(std::shared_ptr<const CueEvent> event) {
  ...
  num_segments_before_last_cue_ = current_segment_index_ + 1;
  return Status::OK;
}
```

In `EndSegmentIfStarted` function:

```c++
Status ChunkingHandler::EndSegmentIfStarted() const {
  ...
  segment_info->segment_index = current_segment_index_ + num_segments_before_last_cue_;
  return DispatchSegmentInfo(kStreamIndex, std::move(segment_info));
}
```

[1] See the code at line 106: ```timestamp - cue_offset_```

`ChunkingHandler::OnFlushRequest` does not need to be updated as it is actually called only once in the end.